### PR TITLE
Move action editing to modal

### LINKED ISSUE #26347: [P1] Move action editor to a modal, explain what it does and how to use it
[Design](https://www.figma.com/file/rbjE1U9kb6IhJQNBY08PAz/Data-Apps-Exploration?node-id=1811%3A29154)

A lot of it is already done.

### PR DESCRIPTION
**Pull request by iethree**
## Description

- moves action creation and editing to modal
- removes action creation from global new menu
- actions can be created or edited from the model detail page or from the action-adding sidebar
- better explainer text for action editor and a snazzy get-sample-code button

## Screenshot
Screen Shot 2022-11-09 at 1 09 39 PM
actionInAction

resolves https://github.com/metabase/metabase/issues/26347

### CODE REVIEW

File: frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionDetails.tsx [hunk 1]
```diff
@@ -60,13 +78,24 @@ function ModelActionDetails({
...
+              borderless
               disabled={isImplicitAction(action)}
+              onClick={
+                !isImplicitAction(action)
+                  ? () => {
+                      setEditingActionId(action.id);
+                      toggleIsActionCreatorVisible();
+                    }
+                  : undefined
+              }
```
Comment (kulyk) [inline frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionDetails.tsx:98]: I personally find this `condition ? () => {} : undefined` kind of expressions a bit hard to read sometimes. Should we maybe always pass a function, but do the check inside? Like:

```jsx
onClick={() => {
  if (isExplicitAction(action)) {
    setEditingActionId(action.id);
    toggleIsActionCreatorVisible();
  }
}}
```

File: frontend/src/metabase/writeback/components/ActionCreator/ActionCreator.tsx [hunk 2]
```diff
@@ -86,34 +96,61 @@ function ActionCreatorComponent({
...
+    onClose?.();
   };
 
   const handleClose = () => setShowSaveModal(false);
 
   const isNew = !actionId && !(question.card() as SavedCard).id;
 
+  const handleExampleClick = () => {
+    const sampleQuery =
+      "UPDATE my_table_name\nSET my_column_name = {{ my_new_value }}\nWHERE id = {{ my_primary_key }}";
```
Comment (kulyk) [inline frontend/src/metabase/writeback/components/ActionCreator/ActionCreator.tsx:108]: Random idea: maybe use a query that'd act on a sample database as an example? It'd be actually runnable, so it'd be easier to explore how things work 🤷‍♂️

File: frontend/src/metabase/writeback/components/ActionCreator/ActionCreator.tsx [hunk 3]
```diff
@@ -86,34 +96,61 @@ function ActionCreatorComponent({
...
+                  setQuestion={setQuestion}
+                />
+              </EditorContainer>
+              <ModalActions>
+                <Button onClick={onClose} borderless>
+                  Cancel
+                </Button>
+                <Button primary onClick={() => setShowSaveModal(true)}>
+                  Save
+                </Button>
```
Comment (kulyk) [inline frontend/src/metabase/writeback/components/ActionCreator/ActionCreator.tsx:139]: "Cancel" and "Save" here should be translated

File: frontend/src/metabase/writeback/components/ActionCreator/FormCreator/EmptyFormPlaceholder.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+export const EmptyFormPlaceholder = ({
+  onExampleClick,
+}: {
+  onExampleClick: () => void;
+}) => (
+  <EmptyFormPlaceholderWrapper>
+    <IconContainer className="text-brand p2 relative inline-block">
+      <Icon name="sql" size={62} />
+      <TopRightIcon name="insight" size={24} />
+    </IconContainer>
```
Comment (kulyk) [inline frontend/src/metabase/writeback/components/ActionCreator/FormCreator/EmptyFormPlaceholder.tsx:23]: Can we avoid CSS class usage here?

**Review by kulyk — CHANGES_REQUESTED** [review]
Looks great overall! Blocking so we don't miss the i18n fix

When playing around with the modal on a data app page, I felt it'd be cool to automatically 
add a new action form/button to the page once I create a new action. If I'm creating a new action from a page, I suppose I want it to appear on my page eventually

File: frontend/src/metabase/writeback/components/ActionCreator/FormCreator/EmptyFormPlaceholder.tsx (hunk 4 above)
Comment (iethree) [reply frontend/src/metabase/writeback/components/ActionCreator/FormCreator/EmptyFormPlaceholder.tsx:23]: 🤦  thanks, this is actually already in emotion, I just didn't remove the classes.

File: frontend/src/metabase/query_builder/components/NativeQueryEditor/utils.ts [hunk 5]
```diff
@@ -15,7 +15,7 @@ const FRACTION_OF_TOTAL_VIEW_HEIGHT = 0.4;
 
 // the query editor needs a fixed pixel height for now
 // until we extract the resizable component
-const FULL_HEIGHT = 500;
+const FULL_HEIGHT = 400;
```
[the remark this answers is not in the document]
Comment (iethree) [reply frontend/src/metabase/query_builder/components/NativeQueryEditor/utils.ts:18]: no, this FULL_HEIGHT constant is just used by actions.

**Review by kulyk — APPROVED** [review]
✨ ✨ ✨

**Pull request merged**